p11-kit/list-objects.c

```diff
@@ -55,10 +57,19 @@
 #define _(x) (x)
 #endif
 
+#define BUFFER_SIZE 128
+
 int
 p11_kit_list_objects (int argc,
 		      char *argv[]);
 
+static inline bool
+is_storage_object (CK_OBJECT_CLASS klass)
```

Comment:
Just out of curiosity: what are the criteria between storage and non-storage objects? If the former means objects with exportable data, I would name this function as "is_exportable_object".

Reply:
From PKCS #11 specification version 3.1: "This is not an object class; hence no CKO_ definition is required. It is a category of object classes with common attributes for the object classes that follow." and the object classes that follow are the same that I have listed in the function.

Also the URI can only be specified for storage objects, so I would rather keep the function named is_storage_object.

Reply:
Sorry I couldn't find any reference of the term "storage object" in the specification, and I still don't understand the difference between your definition of storage and non-storage objects. For example, CKO_X_CERTIFICATE_EXTENSION is *stored* on the token, but cannot be exported individually. Is it storage object or non-storage object?

Reply:
I have seen the term both here https://docs.oasis-open.org/pkcs11/pkcs11-spec/v3.1/pkcs11-spec-v3.1.pdf and here https://datatracker.ietf.org/doc/html/rfc7512
Thats why I decided to use the term storage object.

Reply:
I see; sorry for the oversight.